### PR TITLE
Reload Adapter: New

### PR DESCRIPTION
**Pull request by mafernandez80**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [X] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
bidder: 'reload',
params: {
        plcmID: "prebid_check",
        partID: "part_4",
        opdomID: "0",
        bsrvID: 0
    }
```

- contact email of the adapter’s maintainer: prebid@reload.net
- [X] official adapter submission

- DOC: A link to a [PR #1304](https://github.com/prebid/prebid.github.io/pull/1304)

### CODE REVIEW

File: modules/reloadBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+}
+from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'reload';
+
+const VERSION_ADAPTER = "1.0";
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: [],
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: Please remove aliases property if you are not creating any alias

File: modules/reloadBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+  supportedMediaTypes: [BANNER],
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: Banner is default format. Please remove `supportedMediaTypes` property  if you are only supporting banner

File: modules/reloadBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+  },
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
+
+    var vRequests = [];
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: Use let/const where applicable. We recommend using ES6. Please update at other places as well

File: modules/reloadBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+  },
+
+  /**
+   * Register the user sync pixels which should be dropped after the auction.
+   *
+   * @param {SyncOptions} syncOptions Which user syncs are allowed?
+   * @param {ServerResponse[]} serverResponses List of server's responses.
+   * @return {UserSync[]} The user syncs which should be dropped.
+   */
+  getUserSyncs: function (syncOptions, serverResponses) {
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: Please remove if you are not using getUserSyncs

File: modules/reloadBidAdapter.js [hunk 5]
```diff
@@ -0,0 +1,513 @@
...
+    return [];
+  },
+
+  /**
+   * Register bidder specific code, which will execute if bidder timed out after an auction
+   * @param {data} Containing timeout specific data
+   */
+  onTimeout: function (data) {},
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: Please remove if you are not using onTimeout

File: modules/reloadBidAdapter.js [hunk 6]
```diff
@@ -0,0 +1,513 @@
...
+
+  /**
+   * Register bidder specific code, which will execute if a bid from this bidder won the auction
+   * @param {Bid} The bid that won the auction
+   */
+  onBidWon: function (bid) {},
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: Please remove if you are not using onBidWon

File: modules/reloadBidAdapter.js [hunk 7]
```diff
@@ -0,0 +1,513 @@
...
+
+  onSetTargeting: function (bid) {}
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: Please remove if you are not using onSetTargeting

File: modules/reloadBidAdapter.js [hunk 8]
```diff
@@ -0,0 +1,513 @@
...
+};
+
+/**
+ * Reload Client Tool
+ * @param {json} args
+ */
+
+function ReloadClientTool(args) {
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: What does this function do ?

File: modules/reloadBidAdapter.js [hunk 9]
```diff
@@ -0,0 +1,513 @@
...
+  
+  var that = this;
+  
+
```
Comment (jaiminpanchal27) [inline modules/reloadBidAdapter.js]: Lot of unnecessary line breaks throughout the file. Please remove them

File: test/spec/modules/reloadBidAdapter_spec.js [hunk 10]
```diff
@@ -0,0 +1,300 @@
+import { expect } from 'chai';
+import { spec } from 'modules/reloadBidAdapter';
+import { newBidder } from 'src/adapters/bidderFactory';
```
Comment (jaiminpanchal27) [inline test/spec/modules/reloadBidAdapter_spec.js]: Remove unused imports. Following are not used in your test file anywhere
```
import { newBidder } from 'src/adapters/bidderFactory';
import * as bidderFactory from 'src/adapters/bidderFactory';
import { deepClone } from 'src/utils';
import { config } from 'src/config';
```

File: modules/reloadBidAdapter.js (hunk 8 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: It's a class constructor. It is used to instantiate an interface class that we must use to have a single standard across all our adapters/drivers.

File: modules/reloadBidAdapter.js (hunk 1 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: Fixed

File: modules/reloadBidAdapter.js (hunk 2 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: Fixed

File: modules/reloadBidAdapter.js (hunk 3 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: Fixed

File: modules/reloadBidAdapter.js (hunk 4 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: Fixed

File: modules/reloadBidAdapter.js (hunk 5 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: Fixed

File: modules/reloadBidAdapter.js (hunk 6 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: Fixed

File: modules/reloadBidAdapter.js (hunk 7 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: Fixed

File: modules/reloadBidAdapter.js (hunk 9 above)
Comment (mafernandez80) [reply modules/reloadBidAdapter.js]: Fixed

File: test/spec/modules/reloadBidAdapter_spec.js (hunk 10 above)
Comment (mafernandez80) [reply test/spec/modules/reloadBidAdapter_spec.js]: Fixed

**Comment by jaiminpanchal27** [comment]
@mafernandez80 Circleci failed due to lint errors. You can fix and check on local by running `gulp lint`

**Comment by mafernandez80** [comment]
@jaiminpanchal27 I executed your command but lint ran ok. 
Steps:
1.- Clone repository
2.- npm install
3.- copied our adapter & spec
4.- gulp lint

**Comment by jaiminpanchal27** [comment]
Yes, same happened for me too. Actually on running gulp lint, it autofixex the files. So if you now do git status, it will show you changes to commit. 
To ignore autofixes run `gulp lint --nolintfix`

**Comment by mafernandez80** [comment]
@jaiminpanchal27 I fixed all lint errors

**Comment by mafernandez80** [comment]
@jaiminpanchal27 I see that there is a new release but our adapter has not been included. Is there anything still to be fixed? We really need this ASAP to start our tests. Thnx!

**Comment by jaiminpanchal27** [comment]
@mafernandez80 We release every Tuesday. It will be included in next release. I understand your urgency but one of my colleague had already started work on the release since morning. 

I will test your params on hello world page and let you know if anything more is needed. Code looks good.

**Comment by jaiminpanchal27** [comment]
@mafernandez80 I am not getting valid bid back from your adapter. Here is the response from your endpoint. I used the params given in reloadBidAdapter.md file on Prebid hello_world page.

```
{"id":"5355660366943247","bidid":"fbs-br-d3jbibjgjvpcy1fs","seatbid":[{"bid":[{"id":"fbs-br-stbd-bd-d3jbibjgjvpcy1fu","impid":"2e4c0d6ff12ee4","crid":"2e4c0d6ff12ee4","price":0,"nurl":"","adm":"","ext":{"pcmdata":{},"error":{"msg":"Cannot read property 'fireEvent' of undefined","stack":"TypeError: Cannot read property 'fireEvent' of undefined\n    at fireUnderFilterString_Algo (/bidserver/BidServer/lib/proxy/v2.25/pcm_bundle.js:6323:26)\n    at fireUnderFilterString (/bidserver/BidServer/lib/proxy/v2.25/pcm_bundle.js:6319:9)\n    at _fireEventX (/bidserver/BidServer/lib/proxy/v2.25/pcm_bundle.js:6315:7)\n    at _fireEvent (/bidserver/BidServer/lib/proxy/v2.25/pcm_bundle.js:6329:7)\n    at _fb_PCM (/bidserver/BidServer/lib/proxy/v2.25/pcm_bundle.js:5891:5)\n    at _makePCMCall (/bidserver/BidServer/lib/engine/pcmengine.js:255:16)\n    at Object.callback (/bidserver/BidServer/lib/engine/pcmengine.js:99:15)\n    at Client.memcached.delegateCallback (/bidserver/BidServer/node_modules/memcached/lib/memcached.js:687:12)\n    at Client.rawDataReceived (/bidserver/BidServer/node_modules/memcached/lib/memcached.js:758:22)\n    at Client.BufferBuffer (/bidserver/BidServer/node_modules/memcached/lib/memcached.js:678:12)\n    at Socket.bowlofcurry (/bidserver/BidServer/node_modules/memcached/lib/utils.js:126:15)\n    at Socket.emit (events.js:193:13)\n    at Socket.EventEmitter.emit (domain.js:481:20)\n    at addChunk (_stream_readable.js:295:12)\n    at readableAddChunk (_stream_readable.js:272:13)\n    at Socket.Readable.push (_stream_readable.js:231:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:150:17)"},"plcmID":"4234897234","partID":"part_2","opdomID":0,"bsrvID":0,"banner":{"w":300,"h":250}}}],"seat":"fbs-br-stbd-d3jbibjgjvpcy1ft","group":0}]}
```

**Comment by mafernandez80** [comment]
@jaiminpanchal27 I pasted test parameters when I created this pull request:
...
bidder: 'reload',
params: {
        plcmID: "prebid_check",
        partID: "part_4",
        opdomID: "0",
        bsrvID: 0
    }
...

**Comment by jaiminpanchal27** [comment]
Verified bids back with the params provided in PR description. Please update your test params in `reloadBidAdapter.md` file so I can merge your adapter.

**Comment by mafernandez80** [comment]
@jaiminpanchal27 I updated test params with PR test.

**Pull request merged**